packages/pigeon/lib/objc_generator.dart

```diff
@@ -138,11 +138,52 @@ String _propertyTypeForDartType(NamedType field) {
   }
 }
 
+bool _isNullable(HostDatatype hostDatatype, TypeDeclaration type) =>
+    hostDatatype.datatype.contains('*') && type.isNullable;
+
+void _writeInitializerDeclaration(Indent indent, Class klass,
```

Comment:
That's a good idea.  I do worry about them becoming stale.  What i've seen in other code generators is that they add comments in the generated code that point to the line number of the code that generated it so you can easily see where it is coming from.  I can look at that since everything should be routed through Ident.

Reply:
Actually, they are usually adding comments for line numbers associated with the input, but we could have a flag for it to be the generator's line number

Reply:
I commented below.  That should address the need.

Reply:
> I do worry about them becoming stale.

Any implementation comments could become stale, but the solution is to update them when making changes (and for reviewers to look at that), not to never have implementation comments.

I think the generation flag is potentially interesting for someone actively working on a change, but it's not really useful in a review context. Every time I try to review a method I would have to stop, pull the PR locally, figure out relevant input to try, run it, look at the output, and then map back to the code in the review. In a PR like this that would have to be done many times.

The reason I want this to be a comment is for the same reason as any other implementation comment: to help me as someone *reading* the code (including in a review) to have a framework for what I'm about to read so that I can better understand it.

Reply:
And to be clear, it doesn't have to be a completely comprehensive example in the comment. Just a brief conceptual description of the thing it's making, plus an example or two to give a quick, *in context* demonstration of what the output would look like.

Reply:
I hear you, let me backfill some docstrings and see where that gets us.

Reply:
I backfilled the objc_generator.dart docstrings if you want to give that a look while I start working on the others to make sure I'm on the right path.

Reply:
Done